### ISSUE TITLE
Debugging gulp and using child_process.fork

### ISSUE THREAD

**Issue by cliffmeyers** [jonjaques/gulp-runner#3]
I have the need to debug some of our Gulp-based builds, and we use gulp-runner to execute them. It seems difficult to debug the downstream gulp build because of the use of `child.spawn` here:

https://github.com/jonjaques/gulp-runner/blob/master/index.js#L36

I haven't found a way to get node into debug mode when `child.spawn` is used. I was curious if anyone is aware of a way to do this?

I've experimented with changing gulp-runner to use `child_process.fork` and debugging will work fine, as long as the parent process isn't also set to debug mode. This is due to a Node.js issue and unfortunately the fix for it hasn't landed yet:
bug: https://github.com/nodejs/node/issues/9435
pull: https://github.com/nodejs/node/pull/5025

I was curious if a PR would be accepted that would add two configuration parameters - perhaps `debug=true/false` and `debugPort=number` - which when active would change gulp-runner behavior to use `child_process.fork` instead?

**Comment by cliffmeyers** [comment]
cc @jonjaques

**Comment by jonjaques** [comment]
Hey @cliffmeyers, I have no objection to that! Checkout #5 for how I would like to capture those options. Bonus points if you have any tips for how to run the process w/ `fork`. 😄

**Comment by cliffmeyers** [comment]
@jonjaques yep, I've been fiddling with this the past day or so and finding it a little fickle when run from the test harness. Seems to work fine in my own environment though. I'll check out what you've got and get back to you.
